metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -155,8 +185,30 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         # can't make this method abstract.
         raise NotImplementedError('sources must implement "create"')
 
-    @abstractmethod
+    @property
+    def workunit_processors(self) -> Sequence[Optional[MetadataWorkUnitProcessor]]:
+        """A list of functions that transforms the workunits produced by this source.
+
+        Run in order, first in list is applied first. Be careful with order when overriding.
+        """
+        return [
+            auto_status_aspect,
+            partial(auto_materialize_referenced_tags, active=True),
+            partial(auto_workunit_reporter, self.get_report()),
+            partial(auto_stale_entity_removal, self.stale_entity_removal_handler)
```

Comment:
this should be done in the stateful ingestion source base, not in the source base class

Reply:
Did not mean to include these changes, oops. Specifically wanted this PR to be a straight refactor with no true changes, and why I manually call `auto_workunit_reporter` in each source. Let me remove this + the `stale_entity_removal_handler = None` stuff